### ISSUE TITLE
Collect FB page comments for each Senator and Member of Congress

### ISSUE THREAD

**Issue by gati** [Data4Democracy/just-politics#4]
This issue is the first step for using social media sentiment to assess the popular support (or lack thereof) for a given Member of Congress and Senator, and subsequently testing whether that support score is a useful signal in assessing the vulnerability of their seat.

**Comment by gati** [comment]
The ProPublica API might be helpful: https://propublica.github.io/congress-api-docs/?shell#lists-of-members

**Comment by sidra-ahmad** [comment]
I'd like to help out with this! How far back should I do in terms of collecting Facebook data? Do you have a deadline by which you'd want some preliminary product? Thanks!

**Comment by EmVitz** [comment]
I'd also be interested in helping with this! Like I said on another issue, I've just joined Data for Democracy so am pretty new to all of this. If anyone has specific tasks I could help out with, please let me know.

**Comment by gati** [comment]
@sidra-ahmad That's awesome! I think ideally we'd go back to Jan 1, 2016 for each candidate's page, just to see if there are any useful trends throughout the most active part of the last election cycle. If it's useful, there's an existing project that'll take a bunch of Facebook page ids, get the posts from those pages, and then get all the comments on each post. https://github.com/data4democracy/collect-social I wrote a lot of it, so if it looks helpful and you have questions, just let me know (but no pressure - there's nothing special about the way that module grabs data)

We're shooting to have the bulk of our data in place by the end of May, but there's no hard deadline. :)

Thanks for jumping in and helping with this!

**Comment by gati** [comment]
@EmVitz Yes! I just realized that it'd probably also be helpful to have posts/comments from congressional/senate candidates who lost elections in 2016, so we can compare that dataset to the winners. Unfortunately I think the first step would be to make a list of candidates who list in the House and Senate. Ballotopedia might be a good starting point: https://ballotpedia.org/United_States_House_of_Representatives_elections,_2016 - what do you think?

**Comment by EmVitz** [comment]
@gati Okay great, I can start with that. Just to double check, I'd be looking at the table that has information on all 435 races (it has the heading United States House Margin of Victory and is the 5th table down) on that page, is that right?

**Comment by gati** [comment]
@EmVitz Awesome! As for which table to start with, I should take a step back and say there are two goals: 

1. Understand how people talk about given candidate on social media (in this case, Facebook). One way to do this would be to look at each candidate's Facebook page, and grab all the comments. 

2. Understand how people from a given state (and ideally down to the congressional district) talk about everything. New Knowledge has some pretty advanced tool for understanding people based on what they post online, which we'll donate to this, but right now we don't have state, county, or district-level text to explore. I say all this because that table at the bottom doesn't seem to have candidate's names, but would still be useful to have as a dataset if we wanted to look at (for example) Oklahoma, see how people who live their talk, and then compare that with election outcomes. 

Is that too much information? :) Happy to chat through it on Slack if that'd be helpful.

**Comment by EmVitz** [comment]
@gati Cool, thanks. If we could have a chat on Slack I think that would be good. I've mainly worked with company databases and haven't scraped anything off the web & cleaned it before. Very excited to do it though!

**Comment by gati** [comment]
Sounds good! Just DM when you have a second and we'll figure it out! :)

**Comment by reneejustrenee** [comment]
I'm happy to help with this, too. Maybe we should split the list so we don't duplicate work?

**Comment by sidra-ahmad** [comment]
Hey, Everyone!

Sorry for being MIA; the last couple of weeks have been busy for me. Have we split up the work yet? Are there states/candidates you'd like me to work on?

Thanks!

**Comment by gati** [comment]
I ended up splitting the county-specific FB pages into a separate task (https://github.com/Data4Democracy/just-politics/issues/8) that a couple people are working on. I believe @EmVitz has created a table of candidates that might be helpful. If you wanted to start grabbing FB page URLs, you could use this list here as a jumping off point: https://www.congress.gov/members?q={%22congress%22:%22115%22} :)
